src/event.h

```diff
@@ -0,0 +1,28 @@
+/**
+ * simple library for describing events and waiting for events
+ */
+
+#ifndef TTETRIS_EVENT_H
+#define TTETRIS_EVENT_H
+
+typedef struct ttetris_event TetrisEvent;
```

Comment:
NIT: Having the struct and typedef separate is not something we have done much in our code. We have been defining structs like this entirely in the header files with a typedef. I believe that doing it this way will prevent usage of `struct ttetris_event` in files that import this header? Are there any other differences?

Reply:
Preventing the usage of `struct ttetris_event` was my intent. My thought was to keep the definition opaque and force users to interact with `TetrisEvent`s through the interface defined by the methods in `header.h`. I believe this will make it easier to change the underlying functionality in the future if we want.

I'm not aware of any other differences.

Reply:
I fully agree with this rationale.